src/ast.rs

```diff
@@ -196,13 +317,28 @@ impl PartialEq for ScanArm {
     }
 }
 
+impl DisplayWithContext for ScanArm {
+    fn fmt(&self, f: &mut fmt::Formatter, _ctx: &Context) -> fmt::Result {
+        write!(f, "\"{}\" {{ ... }}", self.regex)
```

Comment:
```suggestion
        write!(f, "{:?} {{ ... }}", self.regex.as_str())
```

to automatically get quoting/escaping of the regex